docs: Add Hubble metrics reference (#11996)
cilium-team/cilium: Documentation/configuration/metrics.rst

Monitoring & Metrics
********************

Cilium and Hubble can both be configured to serve `Prometheus
<https://prometheus.io>`_ metrics. Prometheus is a pluggable metrics collection
and storage system and can act as a data source for `Grafana
<https://grafana.com/>`_, a metrics visualization frontend. Unlike some metrics
collectors like statsd, Prometheus requires the collectors to pull metrics from
each source.

> The Cilium and Hubble metrics can be enabled independently of each other.

Reviewer: ```suggestion
Cilium and Hubble metrics can be enabled independently of each other.
```

Cilium Metrics
==============

> Cilium's metrics provide insight into the state of the Cilium itself, namely

Reviewer: ```suggestion
Cilium metrics provide insights into the state of Cilium itself, namely
```

of the ``cilium-agent`` and ``cilium-operator`` processes. To run Cilium with
Prometheus metrics enabled, deploy it with the
``global.prometheus.enabled=true`` Helm value set.

Cilium metrics are exported under the ``cilium_`` Prometheus namespace.
When running and collecting in Kubernetes they will be tagged with a pod name
and namespace.

Installation
------------

You can enable metrics for ``cilium-agent`` with the Helm value
``global.prometheus.enabled=true``. To enable metrics for ``cilium-operator``,
use ``global.operatorPrometheus.enabled=true``.

.. parsed-literal::

   helm install cilium |CHART_RELEASE| \\
     --namespace kube-system \\
     --set global.prometheus.enabled=true \\
     --set global.operatorPrometheus.enabled=true

The ports can be configured via
``global.prometheus.port`` or ``global.operatorPrometheus.port`` respectively.

When metrics are enabled, all Cilium components will have the annotations to
signal Prometheus whether to scrape metrics:

.. code-block:: yaml

        prometheus.io/scrape: "true"
        prometheus.io/port: "9090"

Hubble Metrics
==============

> While the Cilium metrics allow you to monitor the state Cilium itself,

Reviewer: ```suggestion
While Cilium metrics allow you to monitor the state Cilium itself,
```

> Hubble's metrics on the other hand allow you to monitor the network behavior

Reviewer: ```suggestion
Hubble metrics on the other hand allow you to monitor the network behavior
```

of your Cilium-managed Kubernetes pods with respect to connectivity and security.

Installation
------------

> To deploy Cilium with Hubble's metrics enabled, you need to enable Hubble with

Reviewer: ```suggestion
To deploy Cilium with Hubble metrics enabled, you need to enable Hubble with
```

``global.hubble.enabled=true`` and provide a set of Hubble metrics you want to

> enabled via ``global.hubble.metrics.enabled``.

Reviewer: ```suggestion
enable via ``global.hubble.metrics.enabled``.
```

> Some of the metrics also can be configured with additional options.

Reviewer: ```suggestion
Some of the metrics can also be configured with additional options.
```

    kubectl apply -f \ |SCM_WEB|\/examples/kubernetes/addons/prometheus/monitoring-example.yaml

It will run Prometheus and Grafana in the ``cilium-monitoring`` namespace. If

> you have either enabled the Cilium or the Hubble metrics, they will automatically

Reviewer: ```suggestion
you have either enabled Cilium or Hubble metrics, they will automatically
```
